src/sleep/sleep.c

```diff
@@ -40,21 +40,25 @@ static int write_hibernate_location_info(void) {
                 return log_debug_errno(r, "Unable to find hibernation location: %m");
 
         /* if it's a swap partition, we just write the disk to /sys/power/resume */
-        if (streq(type, "partition"))
-                return write_string_file("/sys/power/resume", device, 0);
-        else if (!streq(type, "file"))
-                return log_debug_errno(EINVAL, "Invalid hibernate type %s: %m",
-                                       type);
+        if (streq(type, "partition")) {
```

Comment:
I think you need an unconditional `return` on this block, otherwise this will fallthrough to `if (!streq(type, "file"))` and end up logging an invalid hibernate and returning -EINVAL... No?

Reply:
ouch, indeed